lib/net/networking_engine.js

```diff
@@ -670,6 +678,23 @@ shaka.net.NetworkingEngine = class extends shaka.util.FakeEventTarget {
     });
   }
 
+  /**
+   * @param {(shaka.extern.RequestContext|undefined)} context
+   * @return {boolean}
+   * @private
+   */
+  allowSwitch_(context) {
+    if (!context) {
```

Comment:
Hm. Maybe it's just because I'm tired when reading this, but it makes me wonder if this would be clearer if instead of exiting early with `true`, we exited early with `false`. After all, this only returns false in one very specific circumstance, so maybe spelling that circumstance out more directly would be better.
E.g.:
```js
if (context) {
  const segment = context.segment;
  const stream = context.stream;
  if (segment && stream && stream.fastSwitching) {
    if (segment.isPartial() && !segment.isLastPartial()) {
      return false;
    }
  }
}
return true;
```

Reply:
Changed!